File: VB/XtraReport1.vb
Imports System
Imports System.Drawing
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace RepStoredProc

    Public Partial Class XtraReport1
        Inherits XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub XtraReport1_BeforePrint(ByVal sender As Object, ByVal e As CancelEventArgs)
            custOrdersDetailTableAdapter.Fill(TryCast(DataSource, NorthwindDataSet).CustOrdersDetail, Convert.ToInt32(Parameters(CStr("OrderId")).Value))
        End Sub
    End Class
End Namespace

File: CS/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;
using RepStoredProc.NorthwindDataSetTableAdapters;

namespace RepStoredProc {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        private void XtraReport1_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            custOrdersDetailTableAdapter.Fill((this.DataSource as NorthwindDataSet).CustOrdersDetail, Convert.ToInt32(Parameters["OrderId"].Value));
        }
    }
}
